syncing/src/main/kotlin/maru/syncing/SyncController.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.syncing
+
+interface SyncController : ChainHeadUpdatedHandler {
+  fun start()
+
+  fun stop()
+
+  fun isBeaconChainSynced(): Boolean
+
+  fun isELSynced(): Boolean
+
+  fun onBeaconSyncComplete(handler: () -> Unit)
+
+  fun onELSyncComplete(handler: () -> Unit)
```

Comment:
I feel like it's wrong to put these methods here. IMO they belong to s separate SyncStatus class, which would be responsible for the state management and access (isBeaconChainSynced, isELSynced). Reg. subscription methods, I'm less clear, whether they should be moved somewhere or not